### PR TITLE
Tests for editing services for mysql, pgsql, proxysql and mongo, rds, azzure

### PR DESCRIPTION
**Pull request by saikumar-vs**
Latest runs with tests 
(https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-tests/detail/pmm2-ui-tests/39721/pipeline)

### CODE REVIEW

**Review by puneet0191 — DISMISSED** [review]
looks good to me, just ensure you have checked with Upgrade tests also once using this custom branch

File: tests/configuration/pages/pmmInventoryPage.js [hunk 1]
```diff
@@ -373,4 +455,130 @@ module.exports = {
...
+  async editRemoteServiceDisplayed(serviceName) {
+    I.waitForElement(this.fields.kebabMenu(serviceName),30);
+    I.wait(10);
+    I.click(this.fields.kebabMenu(serviceName));
+    I.waitForElement(this.fields.editAction,30);
+    I.wait(10);
+    I.click(this.fields.editAction);
+    I.waitForElement(this.fields.editText);
+    I.seeElement(this.fields.editText);
+    switch (serviceName) {
```
Comment (vyarosh) [inline tests/configuration/pages/pmmInventoryPage.js:496]: please remove this long hardcoded switch-case block in favor of generic function argument with doc: functions should accept object of a definite structure eg: `{ serviceName: string, settingsForm: {[formFiled: string]: string } }`

File: tests/configuration/pages/pmmInventoryPage.js [hunk 2]
```diff
@@ -1,15 +1,88 @@
...
+    cluster: 'rds80-cluster new',
+    replicationSet: 'rds80-replication new',
+  },
+  postgresqlInputs: {
+    userName: remoteInstancesHelper.remote_instance.aws.aws_postgresql_12.userName,
+    password: remoteInstancesHelper.remote_instance.aws.aws_postgresql_12.password,
+    environment: 'RDS Postgres new',
+    cluster: 'rdsPostgres-cluster new',
+    replicationSet: 'rdsPostgres-replication new',
+  },
```
Comment (vyarosh) [inline tests/configuration/pages/pmmInventoryPage.js:79]: This is not a part of a page or page object. It's a test data - please remove test related data to the tests file itself or to any test-data-provider file if you would like to store it separately from the test

**Review by vyarosh — CHANGES_REQUESTED** [review]

File: tests/configuration/pages/pmmInventoryPage.js (hunk 1 above)
Comment (saikumar-vs) [reply tests/configuration/pages/pmmInventoryPage.js]: Thanks, addressed

File: tests/configuration/pages/pmmInventoryPage.js (hunk 2 above)
Comment (saikumar-vs) [reply tests/configuration/pages/pmmInventoryPage.js]: Thanks, addressed

File: tests/configuration/pages/pmmInventoryPage.js [hunk 3]
```diff
@@ -18,13 +91,18 @@ module.exports = {
...
+    environment: '$environment-text-input',
     externalExporter: locate('td').withText('External exporter'),
+    editAction: '//span[contains(text(),"Edit")]',
+    editText: '//h3[contains(text(),"Editing")]',
     forceModeCheckbox: locate('$force-field-label'),
     inventoryTable: locate('table'),
     inventoryTableColumn: locate('table').find('td'),
     inventoryTableRows: locate('tr').after('table'),
     inventoryTableRowCount: (count) => locate('span').withText(`${count}`),
+    kebabMenu: (serviceName) => `(${service(serviceName)}/following::span//*[name()='svg' and @class='css-hj6vlq'])[2]`,
```
[the remark this answers is not in the document]
Comment (saikumar-vs) [reply tests/configuration/pages/pmmInventoryPage.js]: Thanks, addressed

File: tests/configuration/pages/pmmInventoryPage.js [hunk 4]
```diff
@@ -18,13 +91,18 @@ module.exports = {
     agentDetailsLabelByText: (label) => locate('[aria-label="Tags"]').find('li').withText(label),
     agentsLink: locate('[role="tablist"] a').withText('Agents').withAttr({ 'aria-label': 'Tab Agents' }),
     agentsLinkOld: locate('a').withText('Agents'),
+    cluster: '$cluster-text-input',
     deleteButton: locate('span').withText('Delete'),
+    environment: '$environment-text-input',
     externalExporter: locate('td').withText('External exporter'),
+    editAction: '//span[contains(text(),"Edit")]',
+    editText: '//h3[contains(text(),"Editing")]',
```
[the remark this answers is not in the document]
Comment (saikumar-vs) [reply tests/configuration/pages/pmmInventoryPage.js]: Thanks, addressed

File: tests/configuration/pages/pmmInventoryPage.js [hunk 5]
```diff
@@ -373,4 +455,130 @@ module.exports = {
...
+    I.fillField(this.fields.replicationSet, serviceParameters.replicationSet);
+  },
+
+  async saveConfirm() {
+    I.click(this.fields.saveButton);
+    I.seeElement(this.fields.savePopupMessage);
+    I.click(this.fields.saveConfirmButton);
+  },
+
+  async checkLabels(serviceParameters) {
```
[the remark this answers is not in the document]
Comment (saikumar-vs) [reply tests/configuration/pages/pmmInventoryPage.js]: Thanks, addressed

File: tests/configuration/pages/pmmInventoryPage.js [hunk 6]
```diff
@@ -373,4 +455,130 @@ module.exports = {
...
+    var labels;
+    labels = `//span[contains(text(),"${serviceParameters.environment}")]`;
+    I.waitForElement(labels, 30);
+    labels = `//span[contains(text(),"${serviceParameters.cluster}")]`;
+    I.waitForElement(labels, 30);
+    labels = `//span[contains(text(),"${serviceParameters.replicationSet}")]`;
+    I.waitForElement(labels, 30);
+  },
+  
+  async editRemoteServiceDisplayed(serviceName) {
```
[the remark this answers is not in the document]
Comment (saikumar-vs) [reply tests/configuration/pages/pmmInventoryPage.js]: this method mainly looks at the new feature edit of services lables now, i have renamed it to editRemoteServiceDisplayed

**Review by vyarosh — DISMISSED** [review]
Looks good to me

**Comment by saikumar-vs** [comment]
> after providing fixes please run ui-tests-matrix and nightly

https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-tests/detail/pmm2-ui-tests/39792/tests

**Pull request merged**